packages/components/src/DateTimePickers/DateTime/datetime-extraction.js

```diff
@@ -217,7 +217,7 @@ function updatePartsOnDateChange(datePickerPayload, time, options) {
 		date: date || dateTextInput,
 		time,
 		datetime,
-		textInput: dateAndTimeToStr(date || dateTextInput, time, options),
+		textInput: dateAndTimeToStr(dateTextInput || date, time, options),
```

Comment:
I am worried about this swap. It looks to me extracted date is more important than text input.
Are you sure about this ?

Reply:
Yes, the swap solve a particular problem: when user manual input an invalid date (eg. '2021-06-xx'), we can still get the user input ('2021-06-xx 12:00'). date will be an invalid Date object so can't get the date string('Invalid Date 12:00').
![image](https://user-images.githubusercontent.com/3214228/121499305-d9a93f80-ca0f-11eb-8a8d-a1c2c85cc040.png)